mne/simulation/evoked.py

```diff
@@ -82,6 +82,18 @@ def simulate_evoked(fwd, stc, info, cov, nave=3., tmin=None, tmax=None,
     evoked = apply_forward(fwd, stc, info)
     if nave < np.inf:
         noise = simulate_noise_evoked(evoked, cov, iir_filter, random_state)
+
+        # Convert snr to nave before deprecation
+        if snr == np.inf:
+            nave = snr
+        elif snr is not None:
+            tmask = _time_mask(evoked.times, tmin, tmax,
+                               sfreq=evoked.info['sfreq'])
```

Comment:
`tmin` and `tmax` should only be used for calculating SNR. SNR is being removed, so tmin/tmax can go too